src/tests/uat/platform_devices_uat.py

```diff
@@ -85,6 +99,30 @@ def search_devices(cb):
     print('Search Devices.................................OK')
 
 
+def facet_devices(cb):
+    """Verify that the SDK returns the same result for Facet Devices as the respective API call"""
+    api_facet = facet_devices_api_call().json()
+
+    query = cb.select(Device).where('')
+    list_facets = query.facets(['policy_id', 'status'])
+
+    api_results = api_facet['results']
+    assert len(api_results) == len(list_facets), \
+        f"Test failed: API call returned {len(api_results)}, but SDK call returned {len(list_facets)}"
+
+    for api_facet, list_facet in zip(api_results, list_facets):
+        assert api_facet['field'] == list_facet.field, \
+            f"Test failed: field name from API {api_facet['field']} differs from SDK {list_facet.field}"
+        assert len(api_facet['values']) == len(list_facet.values_), \
+            f"Test failed: length of values for field {list_facet.field} differs - " \
+            f"API says {len(api_facet['values'])} but SDK says {len(list_facet.values_)}"
+        for vdict, val in zip(api_facet['values'], list_facet.values_):
+            assert vdict == val._info, \
```

Comment:
I'm wondering if it is possible for the API to return the values in a different order in the 2 calls. If it is, this will fail due to different order even if as a whole the responses contain the same values.

Reply:
The ordering appears to be consistent between calls. It's primarily "most popular" (highest total value) first.